roaster-jaxrs/src/main/java/hu/icellmobilsoft/roaster/jaxrs/response/ResponseProcessor.java

```diff
@@ -0,0 +1,231 @@
+import hu.icellmobilsoft.coffee.module.mp.restclient.provider.DefaultLoggerClientRequestFilter;
+import hu.icellmobilsoft.coffee.module.mp.restclient.provider.DefaultLoggerClientResponseFilter;
+import hu.icellmobilsoft.roaster.api.TestException;
+
+/**
+ * Base Response REST handler based on JXRS
+ *
+ * @param <RESPONSE>
+ *            response class (any type)
+ * @author imre.scheffer
+ * @since 0.8.0
+ */
+public abstract class ResponseProcessor<RESPONSE> {
+
+    /**
+     * Base URI config key
+     *
+     * @return config key like "project.service.base.uri"
+     */
+    public abstract String baseUriKey();
+
+    /**
+     * Get value by {@link #baseUriKey()} from microprofile config
+     *
+     * @return value like "http://localhost:8080"
+     */
+    public String baseUri() {
+        return ConfigProvider.getConfig().getValue(baseUriKey(), String.class);
+    }
+
+    /**
+     * HTTP path to call
+     *
+     * @return value like "/test/service/generate/testData"
+     */
+    public abstract String path();
+
+    /**
+     * Expected REST response status code<br>
+     * <br>
+     * Default value is <strong>{@value Status#OK}</strong>
+     */
+    private int expectedStatusCode = Status.OK.getStatusCode();
+
+    /**
+     * Call and get JSON object from HTTP GET method
+     *
+     * @param responseClass
+     *            response class
+     * @param pathParams
+     *            The path parameters.
+     * @return response object cast to responseClass
+     */
+    public RESPONSE getOctetStream(Class<RESPONSE> responseClass, Object... pathParams) {
+        // REST kliens megszerzese
+        ResteasyClient client = (ResteasyClient) ClientBuilder.newClient();
+        ResteasyWebTarget target = client.target(UriBuilder.fromUri(baseUri() + path()).build(pathParams));
+        target.register(MultipartFormDataWriter.class);
+        CDI<Object> cdi = CDI.current();
+        // ez csak MP rest clientnel tud mukodni
+        // target.register(RoasterRestClientBuilderListener.class);
+        target.register(cdi.select(DefaultLoggerClientRequestFilter.class).get());
+        target.register(cdi.select(DefaultLoggerClientResponseFilter.class).get());
+        // GET kuldese
+        Builder builder = target.request().accept(MediaType.APPLICATION_OCTET_STREAM_TYPE);
+        builder = clientBuilderCustomization(builder);
+        Response response = builder.get();
+        return processResponse(response, responseClass);
+    }
+
+    /**
+     * Http JAXRS rest client response customization
+     * 
+     * @param response
+     *            JAXRS rest client response
+     * @param responseClass
+     *            expected response class
+     * @return Parsed response entity
+     */
+    protected RESPONSE processResponse(Response response, Class<RESPONSE> responseClass) {
+        if (response.getStatus() == expectedStatusCode) {
+            return response.readEntity(responseClass);
```

Comment:
Lehet jó lenne egy `responseClass.isAssignableFrom(Response.class)` ág is, hogy a nyers jax-rs Response objektumot is el lehessen kérni. Fájl letöltésnél pl érdekelheti a tesztet, hogy jött-e vissza content-disposition header és akkor nem kell egy ellenőrzés miatt leszámrazni és felülírni a processResponse-t.

Reply:
Igen igazad van, lehetne ezt meg csavarni. Peldaul ugyan azt a hivasokat is bele kellene rakni amit a restassured tud, csak nem akartam jelenleg tobb idot bele tenni, igy is eleg szivas volt kompletizalni tesztekkel es maga a multipart kezelest kiokoskodni. De mostmar megvan az alap, amibe barki beleteheti ezeket. Addig is megoldhato a `processResponse(Response response, Class<RESPONSE> responseClass)` felulirasaval ezeket a dolgokat.